hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/backend/AbstractBackendEntityEnricher.java

```diff
@@ -256,22 +256,35 @@ protected Map<String, org.hypertrace.core.datamodel.AttributeValue> getAttribute
     return attributes;
   }
 
+  // This function was added to enable the backend entity enricher to support updates.
+  // We can also replace this to an update flow if needed.
+  protected abstract Entity mergeBackendEntity(Entity existingEntity, Entity newEntity);
+
   @Nullable
-  private Entity createBackendIfMissing(Entity backendEntity) {
+  private Entity evaluateBackendEntity(Entity backendEntity) {
     RequestContext requestContext = RequestContext.forTenantId(backendEntity.getTenantId());
     try {
       Optional<Entity> backendFromCache =
           entityCache
               .getBackendIdAttrsToEntityCache()
               .get(requestContext.buildContextualKey(backendEntity.getIdentifyingAttributesMap()));
-      return backendFromCache.orElseGet(
-          () -> {
-            Entity result = this.upsertBackend(backendEntity);
-            LOGGER.info("Created backend:{}", result);
-            return result;
-          });
+      if (backendFromCache.isEmpty()) {
+        Entity result = this.upsertBackend(backendEntity);
+        LOGGER.info("Created backend:{}", result);
+        return result;
+      }
+
+      Entity existingEntity = backendFromCache.get();
+      Entity updatedEntity = mergeBackendEntity(existingEntity, backendEntity);
```

Comment:
`updatedEntity` can be null

Reply:
No they cant be. its just in the tests. it was not needed there so didn't put much thought